polar-core/src/kb.rs

```diff
@@ -147,6 +152,34 @@ impl KnowledgeBase {
                 }
             }
         }
+
+        // For every rule type that is *required*, see that there is at least one corresponding
+        // implementation.
+        for rule_type in self.rule_types.required_rule_types() {
```

Comment:
nit: this whole iteration could probably be a `validate_required_rule_types()` method that we call here w/ the `?` operator to propagate errors northward